atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -342,10 +275,6 @@ def _validate_platforms_in_image(self, image):
             self.log.info('Skipping validation of available platforms '
                           'because expected platforms are unknown')
             return
-        if len(expected_platforms) == 1:
```

Comment:
We were not trying to fetch a manifest list for single platform builds before. If we want to keep this behavior, we would want to change the design to compare the v2s2 manifest digest with the v2s2 manifest digest in the koji build metadata. Do we ever save the v2s2 digest in the koji build metadata?

Is there any case where we do not generate the manifest list?

The other approach here would be to save the media type (as in the design). **If** the manifest list is not available, we would then try to compare the digest with the digest of a v2s2 manifest in koji and fail if such digest is either missing or different from what we calculated.

thouhgs?

Reply:
While talking to @rcerven offline, we agreed that, given there is a codepath which can allow images being pushed without a manifest list, we should try to compare the digests of the v2s2 manifests (in koji and in the registry).

I just pushed an update for that.